docs/ovs-pipeline.md

```diff
@@ -324,6 +376,43 @@ In the future this table may support an additional mode of operations, in which
 it will implement kube-proxy functionality and take care of performing
 load-balancing / DNAT on traffic destined to services.
 
+### CnpEgressRuleTable (45)
+
+For this table, you will need to keep mind the CNP [specification](#antrea-network-policy-crd-implementation)
+that we are using.
+
+This table is used to implement the egress rules across all ClusterNetworkPolicies.
+If you dump the flows for this table, you should see something like this:
+```
+1. table=45, priority=64990,ct_state=-new+est,ip actions=resubmit(,70)
+2. table=45, priority=11800,conj_id=2,ip actions=load:0x2->NXM_NX_REG6[],resubmit(,70)
+3. table=45, priority=11800,ip,nw_src=40.0.0.6 actions=conjunction(2,1/3)
+4. table=45, priority=11800,ip,nw_dst=40.0.0.8 actions=conjunction(2,2/3)
+5. table=45, priority=11800,udp,tp_dst=53 actions=conjunction(2,3/3)
+6. table=45, priority=0 actions=resubmit(,50)
+```
+
+Similar to K8s NetworkPolicy implementation, CnpEgressRuleTable also relies on
+the OVS built-in `conjunction` action to implement policies efficiently.
+The above example flows read as follow: if the source IP address is in set
+{40.0.0.6}, and the destination IP address is in the set {40.0.0.8}, and the
+destination TCP port is in the set {53}, then use the `conjunction` action with
+id 2, which goes to [L3ForwardingTable]. Otherwise, go to [EgressRuleTable].
+
+If the `conjunction` action is matched, packets are "allowed" or "dropped"
+based on the `action` field of the CNP rule. If allowed, they are forwarded
+directly to [L3ForwardingTable]. Other packets go to [EgressRuleTable] to be
+evaluated for K8s NetworkPolicy egress rules. If a connection is established,
+its packets go straight to [L3ForwardingTable], with no other match required
```

Comment:
Maybe rephrase this sentence to make it clearer, like "After a connection is established, its following packets go straight..."
We might also add efficiency consideration to match only the first packet.

Reply:
done